ingestion/src/legacy/raw.rs

```diff
@@ -0,0 +1,421 @@
+                    e, offset
+                );
+                continue;
+            };
+
+            if let Err(e) = offset_tx.send(offset).await {
+                metrics::counter!(KAFKA_RAW_FAILURES).increment(1);
+                error!("Failed to send offset: {}", e);
+            };
+            raw_buffer.clear();
+        }
+    });
+
+    'consume_loop: loop {
+        tokio::select! {
+            _ = cancel_token.cancelled() => {
+                info!("Cancellation token triggered");
+                // This will cause db_task to break and return
+                drop(db_tx);
+                break;
+            }
+            Some(Offset { partition, offset }) = offset_rx.recv() => {
+                if let Err(e) = consumer.store_offset(topic, partition, offset) {
+                    metrics::counter!(KAFKA_RAW_FAILURES).increment(1);
+                    error!("failed to mark offset on raw queue: {}", e);
+                }
+            }
+            poll_result = consumer.recv() => {
+                match poll_result {
+                    Err(e) => {
+                        metrics::counter!(KAFKA_RAW_FAILURES).increment(1);
+                        error!("failed to poll raw kafka: {}", Error::Kafka(e));
+                    }
+                    Ok(message) => {
+                        metrics::counter!(KAFKA_RAW_MESSAGES_RECEIVED).increment(1);
+
+                        match message.payload() {
+                            Some(payload) => {
+                                if let Err(e) = 'parse_block: {
+                                    if !is_kldata_message(payload)? {
+                                        // TODO: explain
+                                        continue 'consume_loop;
+                                    }
+                                    // TODO: remove unwrap
+                                    let payload_str = std::str::from_utf8(payload)?.trim();
+
+                                    let offset = Offset { partition:message.partition(), offset: message.offset() };
+
+                                    // TODO: remove clone?
+                                    match parse(payload_str, param_conversions.clone()){
+                                        Ok(data) => db_tx.send((data, offset)).await.unwrap(),
+                                        Err(e) => {
+                                            error!("failed kldata message:\n{:?}", payload_str);
+                                            break 'parse_block Err(e.into())
+                                        },
+                                    }
+
+                                    Ok::<(), Error>(())
+                                } {
+                                    metrics::counter!(KAFKA_RAW_FAILURES).increment(1);
+                                    error!("failed to parse kldata message: {}", e);
+                                }
+                            },
```

Comment:
We don't need the `if let(Err)`?

```suggestion
                            if !is_kldata_message(payload)? {
                                // TODO: explain
                                continue;
                            }
                            // TODO: remove unwrap
                            let payload_str = std::str::from_utf8(payload)?.trim();

                            let offset = Offset { partition:message.partition(), offset: message.offset() };

                            // TODO: remove clone?
                            match parse(payload_str, param_conversions.clone()){
                                Ok(data) => db_tx.send((data, offset)).await.unwrap(),
                                Err(e) => {
                                    error!("failed kldata message:{e}\n{:?}", payload_str);
                                    metrics::counter!(KAFKA_RAW_FAILURES).increment(1);
                                },
                            }
```

Reply:
I guess you'd have to expand everything that can error if you want to get rid of the extra indentation, just a matter of taste. Feel free to resolve

Reply:
This version will kill the task if it encounters an error in parsing.

The point of the block and `if let` is to allow us to `?` all the errors without shortcutting the whole function (they only shortcut the block), so we can log/metric any errors, with only one call, instead of having to put 
```
if let Err(e) = ... {
    error!("error_message");
    metrics::counter!(KAFKA_RAW_FAILTURES).increment(1);
}
```
On everything that can error